backend/src/main/java/com/ethan5/service/UserService.java

```diff
@@ -21,4 +46,61 @@ public String createUser(String id) {
     public User readUser(String id) {
         return repository.findById(id).orElse(null);
     }
+
+    public Playlist getUserTopTracks(String userId, String authHeader) {
+        try {
+            Set<Track> userTopTracks = fetchTopUserSongs(authHeader);
+
+            Playlist playlist = new Playlist();
+            for (Track t : userTopTracks) {
+                Track persistentTrack = trackRepository.findById(t.getId()).orElse(t);
+                saveTrackIfNotExists(persistentTrack);
+                System.out.println(t.getName());
```

Comment:
Better to be using a logger, like Slf4j, instead of `System.out.println`. No big deal though